v2/tools/generator/internal/codegen/pipeline/create_resource_extension_types.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/functions"
+)
+
+// CreateResourceExtensionsStageID is the unique identifier of this stage
+const CreateResourceExtensionsStageID = "createResourceExtensions"
+
+func CreateResourceExtensions(localPath string, idFactory astmodel.IdentifierFactory) Stage {
+	return MakeLegacyStage(
+		CreateResourceExtensionsStageID,
+		"create Resource Extensions for each resource type",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+
+			extendedResourceTypes := make(astmodel.Types)
+			extendedResourceTypesMapping := make(map[astmodel.TypeName][]astmodel.TypeName)
+			resourcePackageRef := make(map[astmodel.TypeName][]astmodel.PackageReference)
```

Comment:
Don't you already have this information in the `extendedResourceTypesMapping`? You do:
```
extendedResourceTypesMapping[name] = append(extendedResourceTypesMapping[name], typeDef.Name())

//putting package references in map for each extension
resourcePackageRef[name] = append(resourcePackageRef[name], typeDef.Name().PackageReference)
```

but instead of using `resourcePackageRef` I think you can get at the same data by just iterating over the `[]astmodel.TypeName` (the value of `extendedResourceTypesMapping`) and calling `.PackageReference` on the `TypeName`?

I think you can just delete this map entirely

Reply:
I agree - and deriving the set of package references from the set of types guarantees consistency.

Reply:
have made the changes above! now looks cleaner and better :)